airbyte-integrations/connectors/source-google-ads/source_google_ads/spec.json

```diff
@@ -67,6 +67,32 @@
         "maximum": 1095,
         "default": 14,
         "examples": [14]
+      },
+      "custom_queries": {
+        "type": "array",
+        "items": {
+          "type": ["object", "null"],
+          "properties": {
+            "query": {
+              "type": "string",
+              "title": "User defined query to build a report by wish",
+              "examples": ["SELECT segments.ad_destination_type, campaign.advertising_channel_sub_type FROM campaign"],
+              "pattern": "(?i)^select(\\s)+[a-zA-Z0-9._,(\\s)-]*(\\s+)from(\\s+)(campaign|customer|ad_group|ad_group_ad)$"
+            },
+            "primary_key": {
+              "type": ["string", "null"],
+              "title": "The unique field to be used as primary key in destination database (if provided)"
+            },
+            "cursor_field": {
+              "type": ["string", "null"],
+              "title": "Field that will be used to track changes during Incremental sync (if provided). Otherwise only Full Refresh sync will be available."
```

Comment:
this should be description, and title should be `Cursor Field`. Same for other fields

Reply:
but there is a regex pattern `(campaign|customer|ad_group|ad_group_ad)`

Reply:
https://developers.google.com/google-ads/api/rest/design/resource-names page shows like available resources. If there are others, I need a full list. The json-schemas were catched by web-debugger while I run queries in UI. If we could have a list of available resources, I could extend the resources lists and add new schemas.

As an alternative I can add additionalProperties if the resource is unknown?

Reply:
I will add more schemas, but I think I will also add hadler for unknown_schema